Heisenberg4269/Trie.cpp

```diff
@@ -0,0 +1,59 @@
+class Trie
+{
+public:
+    struct tnode
+    {
+        vector <tnode*> child;
+        bool isword;
+    };
+    tnode *root;
+    int vocabLength;
+
+    Trie(int vocab = 26)
+    {
+        vocabLength = vocab;
+        root = makeTnode();
+    }
+
+    tnode* makeTnode()
+    {
+        tnode *newTnode = new tnode;
+        newTnode->child.resize(vocabLength);
+        for(int i = 0; i<vocabLength; i++)
+        {
+            newTnode->child[i] = NULL;
+        }
+        newTnode->isword = false;
+        return newTnode;
+    }
+
+    void insert(string str)
+    {
+        tnode *temp = root;
+        for(int i = 0; i<str.size(); i++)
+        {
+            int key = str[i] - 'a';
+            if(temp->child[key] == NULL)
+            {
+                temp->child[key] = makeTnode();
+            }
+            temp = temp->child[key];
+        }
+        temp->isword = true;
+    }
+
+    bool search(string str)
+    {
+        tnode *temp = root;
+        for(int i = 0; i<str.size(); i++)
```

Comment:
Space

Reply:
space operator space. For example -
`a < b`  
`a = b`  
`a <= b`  
`x < y ? 1 : 0`